images/base/files/usr/local/bin/entrypoint

```diff
@@ -372,6 +372,13 @@ enable_network_magic(){
       sed -i "s#${old_ipv4}#${curr_ipv4}#" /etc/kubernetes/manifests/*.yaml || true
       # this is no longer required with autodiscovery
       sed -i "s#${old_ipv4}#${curr_ipv4}#" /var/lib/kubelet/kubeadm-flags.env || true
+      # controller-manager and scheduler connect to local API endpoint. If the Node's IP changes, server address in
+      # kubeconfig should be updated. However, the server certificate isn't valid for the new IP. We update it to
+      # loopback address as it's an alternative address of the certificate.
+      if [[ "${old_ipv4}" != "${curr_ipv4}" ]]; then
```

Comment:
we are unconditionally changing the IP, do we need the if?

Reply:
I added this check because if docker restart doesn't change the Node's IP, it can continue working with the original IP. I'm not sure which is better: use the trick when it has to, or use it unconditionally as long as the container is restarted. Would like to hear your opinion.

Reply:
yeah, that sounds safer, we don't change the current behavior